libraries/Intellisense/FileSystem/RootedPath.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Intellisense.FileSystem;
+
+/// <summary>
+/// Guarantees that the path either starts with a directory separator (i.e. /) or a drive letter followed by a volume separator (i.e. C:).
+/// </summary>
+internal readonly record struct RootedPath
+{
+    public string Value { get; }
+    private RootedPath(string path)
+    {
+        Value = path;
+    }
+
+    public static RootedPath? Create(string path)
+    {
+        if (!Path.IsPathRooted(path))
+        {
+            return null;
+        }
+
+        return new RootedPath(path);
+    }
+
+    public static RootedPath CreateOrThrow(string path)
+    {
+        if (Create(path) is { } res)
+        {
+            return res;
+        }
+
+        throw new ArgumentException($"Attempted to create {nameof(RootedPath)} from {path} but it was not rooted or it was an invalid path.");
+    }
+
+    public bool IsRootDirectory()
+    {
+
+        if (Value[0].IsDirectorySeparator())
```

Comment:
another corner-case to consider (not required in this PR unless you are particularly keen) is that UNC paths can start with double-slashes (in either direction!)...
```
ls //cinderella/c
ls \\cinderella\c
```

Reply:
👍 It's frustrating that while the file API is quite happy to use UNC paths, there doesn't seem to be very good support for detecting/parsing them in the Path api.... best approach I could find elsewhere was to write my own IsRooted variant that checked for double slash